### ISSUE TITLE
Sent PGP-encrypted messages cannot be decrypted by other clients

### ISSUE THREAD

**Issue by m-wynn** [boothj5/profanity#775]
Sending a PGP-encrypted message with Profanity does not allow it to be decrypted by the sender's other clients.

## Steps to reproduce

1. Open another XMPP client which supports Carbons and PGP, for example Gajim (GTK) or Conversations (for Android).
2. Start a PGP-encrypted chat with another user.
3. Open Profanity
4. Enable Carbons
5. Start a PGP-encrypted chat with that same user in Profanity.
6. Send an encrypted message in Profanity.

## Expected Results:

The sent message should appear in other clients.  Currently, if you were to send a message in Gajim or Conversations, they would appear in all of the sender's other clients (including Profanity). I imagine that the messages are encrypted with both the sender's and receiver's publickey.

## Actual Results

The message doesn't appear, or an error that no private key exists to decrypt the message occurs.

**Comment by boothj5** [comment]
I'm probably missing something since you report that other clients can do this, but I can't see how it would work, for example:

Alice is logged in with Client A1 and A2
Bob is logged in with Client B1 and B2  

1) Alice (A1) encrypts message with Bob's public key
2) Alice (A1) sends to Bob
3) Bob (B1) decrypts with Bob's private key

4) Carbon of incoming message is sent to Bob (B2)
5) Bob (B2) decrypts with Bob's private key

6) Carbon of outgoing message is sent to Alice (A2)
7) **Alice (A2) doesn't have Bob's private key to decrypt message**

The only way step 7 could decrypt is if either Alice has Bob's private key (which obviously can't happen), or the message was encrypted with Alice's public key, but then Bob wouldn't be able to decrypt it in steps 3 and 5 unless he has Alice's private key (which again can't happen).

Steps 4 and 6 are handled server side.

A couple of questions:
1) Which version are you using? I've added sending of Carbons for PGP messages in master since the [mailing list post](https://groups.google.com/forum/?hl=en-GB#!topic/profanitydev/sKqQEOZRM7Q). 
2) Do steps 3 and 5 work as expected?

**Comment by m-wynn** [comment]
1. I'm running Profanity, version 0.5.0dev.master.18555ff cloned and built this morning.  I was the person in the mailing list as well.

2. Step 3 definitely works as expected.  I expect 5 is as well, but I will try to figure that out soon.

**Comment by m-wynn** [comment]
I don't fully understand it, but it seems that, in profanity, [keys[1]](https://github.com/boothj5/profanity/blob/master/src/pgp/gpg.c#L609) is never set when encrypting.  Presumably, setting this to the senders publickey should set both the sender and receiver as the recipient in gpgpme.

For some reference,

[Here](https://trac.gajim.org/changeset/7946575b0726b15b09bb16f8f7d6709ed2214ca2) is Gajim's change when implementing this.  It looks like it's mainly just a matter of adding another "--recipient" to GPG.

Additionally, [here](https://github.com/segler-alex/Pidgin-GPG/pull/14/files) is Pidgin's (abandoned?) progress on implementing this.

**Comment by boothj5** [comment]
That sounds like exactly what I'm missing, I wasn't aware you can encrypt a message for more than one recipient with PGP.

I'll add a fix as soon as I get a moment.

Thanks for the help.

**Comment by boothj5** [comment]
I've added a change in master to also encrypt messages using the sender public key.  This should fix the problem but I don't have a setup where I can test at the moment.

Added some logging and it appears to be encrypted for both sender and receiver, but I can't find a way for GPGME to log which was used to decrypt so if you have time to test, I'd be interested to hear if the issue is resolved for you.
